topdown/eval.go

```diff
@@ -2145,13 +2144,13 @@ func (e evalTree) enumerate(iter unifyIterator) error {
 				}
 			}
 		case ast.Object:
-			err := doc.Iter(func(k, _ *ast.Term) error {
-				return e.e.biunify(k, e.ref[e.pos], e.bindings, e.bindings, func() error {
+			ki := doc.KeysIterator()
```

Comment:
This lets us leverage the lazy nature of `lazyObj` for enumerating the tree, too. It can now be done without converting the values.